include/deal.II/base/quadrature_point_data.h

```diff
@@ -0,0 +1,808 @@
+   * <code>
+   * class MyQData : public TransferableQuadraturePointData
+   * {
+   * public:
+   *   unsigned int number_of_values() const
+   *   {
+   *      return 2;
+   *   }
+   *
+   *   // a function to pack scalars into a vector
+   *   void pack_values(std::vector<double> &values) const
+   *   {
+   *     Assert (scalars.size()==2, ExcInternalError());
+   *     scalars[0] = value1;
+   *     scalars[1] = value2;
+   *   }
+   *
+   *   void unpack_values(const std::vector<double> &values)
+   *   {
+   *     Assert (scalars.size() ==2, ExcInternalError());
+   *     value1 = scalars[0];
+   *     value2 = scalars[1];
+   *   }
+   * };
+   * </code>
+   * Note that the order of packing and unpacking has to be the same.
+   *
+   * This class can then be use with CellDataStorage in the following way:
+   * <code>
+   * CellDataStorage<typename Triangulation<dim,dim>::cell_iterator,MyQData> data_storage;
+   * distributed::ContinuousQuadratureDataTransfer<dim,MyQData> data_transfer(FE_Q<dim>(2),QGauss<dim>(3),QGauss<dim>(4));
+   * //...populate data for all active cells in data_storage
+   * //...mark cells for refinement...
+   * data_transfer.prepare_for_coarsening_and_refinement(triangulation,&data_storage);
+   * triangulation.execute_coarsening_and_refinement();
+   * //...initialize quadrature point data on new cells by calling CellDataStorage::reinit()
+   * data_transfer.interpolate();
+   * </code>
+   * This approach can be extended to quadrature point data with Tensors of
+   * arbitrary order, although with a little bit more work in packing and
+   * unpacking of data inside MyQData class.
+   *
+   * @note Currently coarsening is not supported.
+   *
+   * @note During mesh balancing, children of a single cell can be attributed
+   * to different processors. As a corollary, their parent cells can simultaneously
+   * exist on multiple processors.
+   *
+   * @note The functionality provided by this class can alternatively be achieved
+   * using parallel::distributed::SolutionTransfer . However, that would require
+   * the following steps: (i) create an auxiliary DoFHandler with a (discontinuous Galerkin)
+   * FiniteElement which has enough components to represent all data stored at the
+   * quadrature points; (ii) project the data to the FiniteElement space
+   * and thereby store results in global vectors; (iii) use parallel::distributed::SolutionTransfer
+   * to project FE vectors to the new mesh; and (iv) finally project the
+   * data back to the quadrature points on the new mesh via FEValues class.
+   * The ContinuousQuadratureDataTransfer class aims at simplifying the whole
+   * process by only requiring that the quadrature point data class is derived
+   * from the TransferableQuadraturePointData. Everything else will be done automatically.
+   *
+   * @note This class is not well suited for the transfer of discontinuous
+   * quadrature point data. For example for the elasto-plastic constitutive model
```

Comment:
The term "discontinuous" here is a bit awkward because, by definition, the data you describe at the quadrature points is only known at a discrete set of points -- thus, continuity is not a well defined concept. Maybe describe it as follows:
```
 @note This class is not well suited to situations where the values stored at quadrature points
 represent samples from a discontinuous field. An example for such a situation would be where the
  data stored at the quadrature points represents the elastic or plastic state of a material, i.e., a property
  that varies discontinuously within a solid. In such cases, trying to transfer data from the quadrature points
  to a finite element field that is continuous (at least within one cell) will likely yield over and undershoots
  that, once evaluated at a different set of quadrature points (on child or parent cells) results in values that
  will not make much sense.
```